### PR TITLE
pgl: implement full api

### PR DESCRIPTION
**Pull request by SciresM**
Implements the full api for the new pgl service.

Don't merge yet, will also commit service commands that pgl sysmodule needs to use soon.

### CODE REVIEW

File: nx/include/switch/services/pgl.h [hunk 1]
```diff
@@ -7,6 +7,35 @@
 #pragma once
 #include "../types.h"
 #include "../sf/service.h"
+#include "../services/ncm_types.h"
+#include "../services/pm.h"
+
+/// LaunchFlag
+typedef enum {
+    PglLaunchFlag_None                                     = 0,
+    PglLaunchFlag_EnableDetailedCrashReport                = (1 << 0),
```
Comment (yellows8) [inline nx/include/switch/services/pgl.h:16]: BIT()

File: nx/source/services/pgl.c [hunk 2]
```diff
@@ -21,10 +22,113 @@ Service* pglGetServiceSession(void) {
...
+static Result _pglCmdInBool(Service* srv, u32 cmd_id, bool inval) {
+    const u8 in = inval;
+    return serviceDispatchIn(srv, cmd_id, in);
+}
+
+static Result _pglCmdOutBool(Service* srv, u32 cmd_id, bool *out) {
+    u8 outval = 0;
+    Result rc = serviceDispatchOut(srv, cmd_id, outval);
+    if (R_SUCCEEDED(rc)) {
+        if (out) *out = (outval != 0);
```
Comment (yellows8) [inline nx/source/services/pgl.c:34]: `outval & 1`, same below.

File: nx/source/services/pgl.c [hunk 3]
```diff
@@ -21,10 +22,113 @@ Service* pglGetServiceSession(void) {
     return &g_pglSrv;
 }
 
+static Result _pglCmdInBool(Service* srv, u32 cmd_id, bool inval) {
```
Comment (fincs) [inline nx/source/services/pgl.c:25]: cmd_id goes last

File: nx/source/services/pgl.c [hunk 4]
```diff
@@ -21,10 +22,113 @@ Service* pglGetServiceSession(void) {
...
+    const u8 in = inval;
+    return serviceDispatchIn(srv, cmd_id, in);
+}
+
+static Result _pglCmdOutBool(Service* srv, u32 cmd_id, bool *out) {
```
Comment (fincs) [inline nx/source/services/pgl.c:30]: Same here

File: nx/source/services/lr.c [hunk 5]
```diff
@@ -135,6 +135,10 @@ Result lrLrRefresh(LrLocationResolver* lr) {
     return serviceDispatch(&lr->s, 9);
 }
 
+Result lrLrEraseProgramRedirection(LrLocationResolver* lr, u64 tid) {
```
Comment (yellows8) [inline nx/source/services/lr.c:138]: Sysver check missing (and update docs).

**Pull request merged**